nexus/db-queries/src/db/datastore/volume.rs

```diff
@@ -1794,4 +2215,211 @@ mod tests {
         db.cleanup().await.unwrap();
         logctx.cleanup_successful();
     }
+
+    #[tokio::test]
+    async fn test_volume_replace_region() {
+        let logctx = dev::test_setup_log("test_volume_replace_region");
+        let log = logctx.log.new(o!());
+        let mut db = test_setup_database(&log).await;
+        let (_opctx, db_datastore) = datastore_test(&logctx, &db).await;
+
+        // Insert four Region records (three, plus one additionally allocated)
+
+        let volume_id = Uuid::new_v4();
+        let new_volume_id = Uuid::new_v4();
```

Comment:
The `new_volume_id` for the new region that's allocated to replace the target region is meant to be passed to a volume delete saga after the replacement has completed, and it stores only one target (which would be the replaced region). It's a synthetic volume in that it won't ever correspond to a user visible object (Disk, Snapshot, etc), and it will never be checked out and passed to a `construct`.

The Volume that corresponds to the target region to be replaced is modified in-place (by `volume_replace_region`) so that anything referring to that Volume doesn't need to be modified.

Reply:
That's a great explanation. This was probably my biggest question on the whole PR. Thanks James!

It's almost certainly too large of a change to make here, but I'm wondering if we could add a new type like `SyntheticVolume(new_volume_id)` to clarify this and prevent accidents in the future. Possibly worthy of an issue.